src/rpcq-python.lisp

```diff
@@ -0,0 +1,287 @@
+;;;; rpcq-python.lisp
+;;;;
+;;;; Authors: Nikolas Tezak, Eric Peterson
+
+(in-package #:rpcq)
+
+
+(defparameter *python-types*
+  '(:string "str"
+    :bytes "bytes"
+    :float "float"
+    :integer "int"
+    :bool "bool"
+    :map "Dict"
+    :list "List"
+    :any "Any"))
+
+(defparameter *python-instance-check-types*
+  '(:string "basestring"
+    :bytes "bytes"
+    :float "float"
+    :integer "int"
+    :bool "bool"
+    :map "dict"
+    :list "list"
+    :any "object"))
+
+
+(defun python-instance-check-type (field-type)
+  (let ((*python-types* *python-instance-check-types*))
+    (python-type field-type)))
+
+(defun python-argspec-default (field-type default)
+  "Translate DEFAULT values for immutable objects of a given
+FIELD-TYPE to python."
+  (case field-type
```

Comment:
see comment below about the `cons` case. This could similarly be changed to a giant `typecase`

```
(typecase field-type
  ((eql :string) ...)
  ...
  ((cons (eql :list)) ...)
  (otherwise ...))
```